### PR TITLE
IPAM doesn't honor the pool provided in AutoAssignArgs + UT changes 

### LINKED ISSUE #244: IPAM doesn't honor the pool provided in AutoAssignArgs
**Steps to reproduce:**
- create 2 pools (`pool1`, and `pool2`) 
- AutoAssign one IP from `host-A` from `pool1`
- AutoAssign one IP from `host-A` from `pool2`
You will notice you still get an IP assigned from `pool1` event when you request the IP from `pool2`

### PR DESCRIPTION
**Pull request by gunjan5**
Fixes #244

### CODE REVIEW

File: lib/client/ipam_block_reader_writer.go [hunk 1]
```diff
@@ -55,7 +55,9 @@ func (rw blockReaderWriter) getAffineBlocks(host string, ver ipVersion, pool *cn
 	ids := []cnet.IPNet{}
 	for _, o := range datastoreObjs {
 		k := o.Key.(model.BlockAffinityKey)
-		ids = append(ids, k.CIDR)
+		if reflect.DeepEqual(k.CIDR.IPNet, pool.IPNet) {
```
Comment (matthewdupre) [inline lib/client/ipam_block_reader_writer.go]: This is wrong, we should be checking the block is in the pool, not equal to.  Test also needs correcting, as discussed.

**Comment by matthewdupre** [comment]
(This is out of scope for 2.0-beta)

**Comment by gunjan5** [comment]
@matthewdupre updated the tests as discussed with @caseydavenport. Tests fail without the fix now, and pass with the fix.  PTAL

File: lib/client/ipam_test.go [hunk 2]
```diff
@@ -370,3 +482,26 @@ func assignIPutil(ic client.IPAMInterface, assignIP net.IP, host string) {
...
+func getAffineBlocks(host string) []cnet.IPNet {
+	opts := model.BlockAffinityListOptions{Host: host, IPVersion: 4}
+	compatClient, err := backend.NewClient(api.ClientConfig{BackendType: etcdType, BackendConfig: &etcdConfig})
+
+	datastoreObjs, err := compatClient.List(opts)
+	if err != nil {
+		if _, ok := err.(cerrors.ErrorResourceDoesNotExist); ok {
+			log.Printf("No affined blocks found")
+		} else {
+			fmt.Printf("Error getting affine blocks: %s", err)
```
Comment (matthewdupre) [inline lib/client/ipam_test.go]: This should be a hard error; since it's test code.

Comment (gunjan5) [inline lib/client/ipam_test.go]: done!

**Pull request merged**